lib/logstash/inputs/jms.rb

```diff
@@ -0,0 +1,285 @@
+# encoding: utf-8
+require "logstash/inputs/base"
+require "logstash/inputs/threadable"
+require "logstash/namespace"
+
+# Read events from a Jms Broker. Supports both Jms Queues and Topics.
+#
+# For more information about Jms, see <http://docs.oracle.com/javaee/6/tutorial/doc/bncdq.html>
+# For more information about the Ruby Gem used, see <http://github.com/reidmorrison/jruby-jms>
+# Here is a config example :
+#	 jms {
+#			include_header => false
+#			include_properties => false
+#			include_body => true
+#			use_jms_timestamp => false
+#			interval => 10
+#			queue_name => "myqueue"
+#			yaml_file => "~/jms.yml"
+#			yaml_section => "mybroker"
+#		}
+#
+#
+class LogStash::Inputs::Jms < LogStash::Inputs::Threadable
+	config_name "jms"
+
+	default :config, "plain"
+
+	# A JMS message has three parts :
+	#	 Message Headers (required)
+	#	 Message Properties (optional)
+	#	 Message Bodies (optional)
+	# You can tell the input plugin which parts should be included in the event produced by Logstash
+	#
+	# Include JMS Message Header Field values in the event
+	config :include_header, :validate => :boolean, :default => true
+	# Include JMS Message Properties Field values in the event
+	config :include_properties, :validate => :boolean, :default => true
+	# Include JMS Message Body in the event
+	# Supports TextMessage and MapMessage
+	# If the JMS Message is a TextMessage, then the value will be in the "message" field of the event
+	# If the JMS Message is a MapMessage, then all the key/value pairs will be added in the Hashmap of the event
+	# BytesMessage, StreamMessage and ObjectMessage are not supported
+	config :include_body, :validate => :boolean, :default => true
+	# Convert the JMSTimestamp header field to the @timestamp value of the event
+	# Don't use it for now, it is buggy
+	config :use_jms_timestamp, :validate => :boolean, :default => false
+
+	# Choose an implementation of the run block. Value can be either consumer, async or thread
+	config :runner, :validate => :string, :required => true, :default => "consumer"
+
+	# Set the selector to use to get messages off the queue or topic
+	config :selector, :validate => :string
+
+	# Initial connection timeout in seconds.
+	config :timeout, :validate => :number, :default => 1000
+
+	# Polling interval.
+	# This is the time sleeping between asks to a consumed Queue.
+	# This parameter has non influence in the case of a subcribed Topic.
+	config :interval, :validate => :number, :default => 10
+
+	# If pub-sub (topic) style should be used or not.
+	# Mandatory
+	config :pub_sub, :validate => :boolean, :default => false
+	# Name of the destination queue or topic to use.
+	# Mandatory
+	config :destination, :validate => :string
```

Comment:
For mandatory fields, adding `:required => true` should be enough, instead of the comment. same for `:pub_sub`